app/controllers/users/confirmations_controller.rb

```diff
@@ -0,0 +1,74 @@
+class Users::ConfirmationsController < Devise::ConfirmationsController
+  skip_before_action :authenticate_user!
+
+  # PUT /resource/confirmation
+  def update
+    with_unconfirmed_confirmable do
+      if @confirmable.has_no_password?
```

Comment:
I guess `@confirmable` is an instance variable defined by Devise and not us hence we lack attr_reader 😢

Reply:
@naliwajek `@confirmable` is set in at L53 and L62.